test/unit/org/apache/cassandra/db/filter/ANNOptionsTest.java

```diff
@@ -95,6 +98,13 @@ public void testParseAndValidate()
         execute("SELECT * FROM %s ORDER BY v ANN OF [1, 1] LIMIT 10 WITH ann_options = {'rerank_k': '-1'}");
         execute("SELECT * FROM %s ORDER BY v ANN OF [1, 1] LIMIT 10 WITH ann_options = {'rerank_k': '-1000'}");
 
+        // Queries that exeed the failure threshold for the guardrail. Specifies a protocol version to trigger
+        // validation in the coordinator.
+        assertInvalidThrowMessage(Optional.of(ProtocolVersion.V5),
```

Comment:
We might want to do a brief test extending `GuardrailTester`, there are many of those in the `org.apache.cassandra.guardrails` package to be used as examples. `GuardrailTester` has some utilities to test the validation of config properties, excluded users, etc. It would also be easier to identify in the 5.0 branch, where we have better machinery for testing guardrails.

Reply:
Thanks for pointing that out, adding a test now.

Reply:
Test added, ptal